wasm/src/main/java/com/dylibso/chicory/wasm/io/InputStreamWasmInputStream.java

```diff
@@ -0,0 +1,139 @@
+package com.dylibso.chicory.wasm.io;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+/**
+ * A WASM input stream based on {@link InputStream}.
+ */
+class InputStreamWasmInputStream extends WasmInputStream {
+
+    private long position;
+    private final BufferedInputStream is;
+    private final ByteBuffer littleBuffer =
+            ByteBuffer.wrap(new byte[8]).order(ByteOrder.LITTLE_ENDIAN);
+
+    InputStreamWasmInputStream(final InputStream is) {
+        this.is =
+                is instanceof BufferedInputStream
+                        ? (BufferedInputStream) is
+                        : new BufferedInputStream(is);
+    }
+
+    public long position() {
+        return position;
+    }
+
+    public void skip(long count) {
+        // todo: java 12+: is.skipNBytes(count);
+        if (count <= 0) {
+            return;
+        }
+
+        try {
+            long t;
+            while (count > 0) {
+                t = is.skip(count);
+                if (t == 0) {
+                    if (is.read() == -1) {
+                        throw new WasmEOFException();
+                    }
+                    t = 1;
+                }
+                count -= t;
+                position += t;
+            }
+        } catch (IOException e) {
+            throw new WasmIOException(e);
+        }
+    }
+
+    public int rawByteOpt() {
+        try {
+            int res = is.read();
+            if (res >= 0) {
+                position++;
+                return res & 0xff;
+            } else {
+                return -1;
```

Comment:
Does `-1` have a special meaning?

Reply:
Yes. Please read the documentation of the superclass method; as with `InputStream`, this method returns a byte value between 0-255 inclusive, or otherwise returns -1 if the end of file is reached.

Reply:
Commented above.